nimble/host/services/audio/pacs/src/ble_audio_svc_pacs.c

```diff
@@ -0,0 +1,428 @@
+
+    return rc == 0 ? 0 : BLE_ATT_ERR_INSUFFICIENT_RES;
+}
+
+static int
+ble_svc_audio_pacs_sup_audio_ctx_read_access(struct ble_gatt_access_ctxt
+                                             *ctxt)
+{
+    int rc;
+
+    rc = os_mbuf_append(ctxt->om, &ble_svc_audio_pacs_sup_sink_contexts,
+                        sizeof ble_svc_audio_pacs_sup_sink_contexts);
+    if (rc) {
+        return BLE_ATT_ERR_INSUFFICIENT_RES;
+    }
+
+    rc = os_mbuf_append(ctxt->om, &ble_svc_audio_pacs_sup_source_contexts,
+                        sizeof ble_svc_audio_pacs_sup_source_contexts);
+    return rc == 0 ? 0 : BLE_ATT_ERR_INSUFFICIENT_RES;
+}
+
+static int
+ble_svc_audio_pacs_access(uint16_t conn_handle, uint16_t attr_handle,
+                          struct ble_gatt_access_ctxt *ctxt, void *arg)
+{
+    uint16_t uuid16 = ble_uuid_u16(ctxt->chr->uuid);
+    int rc;
+
+    switch (uuid16) {
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_PAC:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_sink_pac_read_access(ctxt);
+        } else {
+            assert(0);
+            rc = BLE_ATT_ERR_UNLIKELY;
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SINK_AUDIO_LOCATIONS:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_sink_audio_loc_read_access(ctxt);
+        } else {
+            rc = BLE_ATT_ERR_REQ_NOT_SUPPORTED;
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SOURCE_PAC:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_source_pac_read_access(ctxt);
+        } else {
+            assert(0);
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SOURCE_AUDIO_LOCATIONS:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_source_audio_loc_read_access(ctxt);
+        } else {
+            rc = BLE_ATT_ERR_REQ_NOT_SUPPORTED;
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_AVAILABLE_AUDIO_CONTEXTS:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_avail_audio_ctx_read_access(conn_handle,
+                                                                ctxt);
+        } else {
+            assert(0);
+        }
+        return rc;
+    case BLE_SVC_AUDIO_PACS_CHR_UUID16_SUPPORTED_AUDIO_CONTEXTS:
+        if (ctxt->op == BLE_GATT_ACCESS_OP_READ_CHR) {
+            rc = ble_svc_audio_pacs_sup_audio_ctx_read_access(ctxt);
+        } else {
+            assert(0);
+        }
+        return rc;
+    default:
+        assert(0);
+    }
+}
+
+int
+ble_svc_audio_pacs_set_sink_audio_loc(uint32_t audio_locations)
+{
+    ble_svc_audio_pacs_sink_audio_locations = audio_locations;
+
+    return 0;
+}
+
+int
+ble_svc_audio_pacs_set_source_audio_loc(uint32_t audio_locations)
+{
+    ble_svc_audio_pacs_source_audio_locations = audio_locations;
+
+    return 0;
+}
```

Comment:
As the notifications are unsupported, please disallow those functions to be called again.